test/test_runner.go

```diff
@@ -212,7 +214,18 @@ func (r *TestRunner) RunTest(script string, funcName string) (result *Result, er
 		})
 	}()
 
-	_, inter, err := r.parseCheckAndInterpret(script)
+	program, inter, err := r.parseCheckAndInterpret(script)
+	if err != nil {
+		return nil, err
+	}
+
+	err = r.runTestInit(inter)
+	if err != nil {
+		return nil, err
+	}
+
+	script = replaceImports(program.Program, script)
+	_, inter, err = r.parseCheckAndInterpret(script)
```

Comment:
Having replaced the imports, we parse / check & interpret the new test script, to load the correct contracts & their types.

Reply:
In order to be able to replace the imports from `StringLocation` to `AddressLocation`, we first have to parse / check / interpret the entire program, so that we have created a `blockchain` and an `account` where the desired contracts will be deployed.
That is the reason I am doing it twice, because there's no way to know before-hand, with what address to replace the import paths. Let me know if I misunderstood something here :pray:

Reply:
oh sorry, I suggested this assuming we read the address from `flow.json`, as in https://github.com/onflow/cadence-tools/pull/210#discussion_r1321721640. I should have been more clear about it, my bad.

So assuming the above, we could first parse, and replace the imports in the AST, and then use that re-written AST for the checking and interpreting. So the idea is, during type-checking, we load the contract from the source code/ file. During interpreting, when `init` is run, the contract value is initialized.

I can give this a try maybe later today.

Reply:
I have fixed this in https://github.com/onflow/cadence-tools/pull/210/commits/b1a58de3683c2494f2603509c7ca10fde182ab24.
Developers will have to use something like the following, in `flow.json`:
```json
"contracts": {
  "FooContract": {
    "source": "contracts/FooContract.cdc",
    "aliases": {
      "testing": "0x0000000000000007"
    }
  }
}
```